dev-lang/nqp/nqp-2022.12.ebuild

```diff
@@ -0,0 +1,147 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit java-pkg-opt-2 multibuild
+
+if [[ ${PV} == "9999" ]]; then
+	EGIT_REPO_URI="https://github.com/perl6/${PN}.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/perl6/${PN}/releases/download/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Not Quite Perl, a Raku bootstrapping compiler"
+HOMEPAGE="https://rakudo.org"
+
+LICENSE="Artistic-2"
+SLOT="0"
+IUSE="doc clang java +moar test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="|| ( java moar )"
+
+CDEPEND="java? (
+		dev-java/asm:4
+		dev-java/jna:4
+	)
+	moar? ( ~dev-lang/moarvm-${PV}[clang=] )"
+RDEPEND="${CDEPEND}
+	java? ( >=virtual/jre-1.9 )"
+DEPEND="${CDEPEND}"
+BDEPEND="${CDEPEND}
+	clang? ( sys-devel/clang )
+	java? ( >=virtual/jdk-1.9 )
+	dev-lang/perl"
+
+pkg_pretend() {
+	if has_version dev-lang/rakudo || has_version dev-lang/nqp; then
+		ewarn "NQP is known to fail compilation/installation with Rakudo and/or NQP"
+		ewarn "already being installed. So if it fails, try uninstalling both"
+		ewarn "dev-lang/nqp and dev-lang/rakudo, then do a new installation."
+		ewarn "(see Bug #584394)"
+	fi
+}
+
+src_prepare() {
+	MULTIBUILD_VARIANTS=()
+	use moar && MULTIBUILD_VARIANTS+=( moar )
+	use java && MULTIBUILD_VARIANTS+=( jvm )
+
+	multibuild_copy_sources
+
+	# This will pull in conditional java_prepare
+	default
```

Comment:
But… how? `default` calls `default_src_prepare`, not anything java-specific.

Reply:
I didn't write it. Patrick did. I merely copied his ebuild. I will figure out what to do about this.

Reply:
My contribution to rakudo ecosystem is to add install-raku.dist and rakudo.eclass and rakudo.eclass test dependencies. I tried not to touch rakudo, nqp, and moarvm.

Reply:
Should I just replace `default` with `java-pkg-opt-2_src_prepare`?

Reply:
I suppose that was the intent.

Reply:
Replaced `default` with `java-pkg-opt-2_src_prepare`.